modin/numpy/arr.py

```diff
@@ -1020,74 +1106,94 @@ def prod(
         if self._ndim == 1:
             if axis == 1:
                 raise numpy.AxisError(1, 1)
-            result = self._query_compiler.astype(
-                {col_name: out_dtype for col_name in self._query_compiler.columns}
+            result = self._query_compiler
+            if isinstance(where, array):
+                result = where.where(self, 1)._query_compiler
+            result = result.astype(
+                {col_name: out_dtype for col_name in result.columns}
             ).prod(axis=0)
-            if initial is not None:
-                result = result.mul(initial)
+            result = result.mul(initial)
```

Comment:
what would happen if `initial` is None?

Reply:
I think it would fail, but we set `initial` to 1 if it is `None` at the top, so it doesn't matter.